core/src/main/java/jenkins/security/UpdateSiteWarningsMonitor.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * <ul>
+ *   <li>Intersection of applicable and inapplicable
+ *   <li>Intersection of active and hidden
+ *   <li>Intersection of active and inapplicable
+ * </ul>
+ *
+ * @since TODO
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class UpdateSiteWarningsMonitor extends AdministrativeMonitor {
+    @Override
+    public boolean isActivated() {
+        return !getActiveCoreWarnings().isEmpty() || !getActivePluginWarningsByPlugin().isEmpty();
+    }
+
+    public List<UpdateSite.Warning> getActiveCoreWarnings() {
+        List<UpdateSite.Warning> CoreWarnings = new ArrayList<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!warning.isCoreWarning()) {
+                // this is not a core warning
+                continue;
+            }
+            CoreWarnings.add(warning);
+        }
+        return CoreWarnings;
+    }
+
+    public Map<PluginWrapper, List<UpdateSite.Warning>> getActivePluginWarningsByPlugin() {
+        Map<PluginWrapper, List<UpdateSite.Warning>> activePluginWarningsByPlugin = new HashMap<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!warning.isPluginWarning()) {
+                // this is not a plugin warning
+                continue;
+            }
+
+            String pluginName = warning.component;
+
+            PluginWrapper plugin = Jenkins.getInstance().getPluginManager().getPlugin(pluginName);
+
+            if (!activePluginWarningsByPlugin.containsKey(plugin)) {
+                activePluginWarningsByPlugin.put(plugin, new ArrayList<UpdateSite.Warning>());
+            }
+            activePluginWarningsByPlugin.get(plugin).add(warning);
+        }
+        return activePluginWarningsByPlugin;
+
+    }
+
+    private Set<UpdateSite.Warning> getActiveWarnings() {
+        ExtensionList<UpdateSiteWarningsConfiguration> configurations = ExtensionList.lookup(UpdateSiteWarningsConfiguration.class);
+        if (configurations.isEmpty()) {
+            return Collections.emptySet();
+        }
+        UpdateSiteWarningsConfiguration configuration = configurations.get(0);
+
+        HashSet<UpdateSite.Warning> activeWarnings = new HashSet<>();
+
+        for (UpdateSite.Warning warning : configuration.getApplicableWarnings()) {
+            if (!configuration.getIgnoredWarnings().contains(warning.id)) {
+                activeWarnings.add(warning);
+            }
+        }
+
+        return Collections.unmodifiableSet(activeWarnings);
+    }
+
+    /**
+     * Redirects the user to the plugin manager or security configuration
+     */
+    @RequirePOST
+    public HttpResponse doForward(@QueryParameter String fix, @QueryParameter String configure) {
+        if (fix != null) {
+            return HttpResponses.redirectViaContextPath("pluginManager");
+        }
+        if (configure != null) {
+            return HttpResponses.redirectViaContextPath("configureSecurity");
+        }
+
+        // shouldn't happen
```

Comment:
It's nothing bad, just an URL accessed without one of the two possible location parameters.